core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala

```diff
@@ -285,6 +297,34 @@ class HealthMonitor private (val slickDataSource: SlickDataSource,
     samDAO.getStatus()
   }
 
+  private def checkBPM: Future[SubsystemStatus] = {
+    logger.debug("Checking Billing Profile Manager...")
+    Future(
+      SubsystemStatus(
+        billingProfileManagerDAO.getStatus().isOk,
+        Option(billingProfileManagerDAO.getStatus().getSystems).map { subSystemStatuses =>
+          val messages = for {
+            (subSystem, subSystemStatus) <- subSystemStatuses.asScala
+            message <- Option(subSystemStatus.getMessages).map(_.asScala).getOrElse(Seq("none"))
+          } yield s"$subSystem: (ok: ${subSystemStatus.isOk}, message: $message)"
+          messages.toList
+        }
+      )
+    )
+  }
+
+  private def checkWSM: Future[SubsystemStatus] = {
+    logger.debug("Checking Workspace Manager...")
+    Future(
+      try {
+        workspaceManagerDAO.getStatus()
+        OkStatus
```

Comment:
Do we need line 321? I think we just want to return line 320 (with the exception error handling to map to failed).

Reply:
Agreed - you can use `recover` to return a `failedStatus` rather than explict `try-catch`

Reply:
I included `OkStatus` in line 321 because `getStatus()` for WSMDao is a void method (it returns an Ok status if `getStatus()` doesn't throw any ApiExceptions). 
I currently have this, but I get an error because getStatus() returns a `Unit` but the method expects `Future[SubsystemStatus]`—I’m not sure which method to use with `recover` so that it returns the correct type: 

 ```
private def checkWSM: Future[SubsystemStatus] = {
    logger.debug("Checking Workspace Manager...")
    Future(
        workspaceManagerDAO.getStatus()
    ).recover{
      case exception: ApiException => failedStatus(s"WorkspaceManager: (ok: false, message: $exception)")
    }
  }
```

Reply:
Future({
  workspaceManagerDAO.getStatus()
  OkStatus
})

Reply:
got it, thank you!